src/vm/tests/costs.rs

```diff
@@ -651,68 +666,52 @@ fn test_tracked_costs(prog: &str) -> ExecutionCost {
 
-    let marf_kv = MarfedKV::temporary();
-    let mut clarity_instance = ClarityInstance::new(false, marf_kv);
-    clarity_instance
-        .begin_test_genesis_block(
-            &StacksBlockId::sentinel(),
-            &StacksBlockHeader::make_index_block_hash(
-                &FIRST_BURNCHAIN_CONSENSUS_HASH,
-                &FIRST_STACKS_BLOCK_HASH,
-            ),
-            &TEST_HEADER_DB,
-            &TEST_BURN_STATE_DB,
-        )
-        .commit_block();
-
-    let mut marf_kv = clarity_instance.destroy();
-
-    let mut store = marf_kv.begin(
-        &StacksBlockHeader::make_index_block_hash(
-            &FIRST_BURNCHAIN_CONSENSUS_HASH,
-            &FIRST_STACKS_BLOCK_HASH,
-        ),
-        &StacksBlockId([1 as u8; 32]),
-    );
+    with_owned_env(epoch, |mut owned_env| {
+        owned_env
+            .initialize_contract(trait_contract_id.clone(), contract_trait)
+            .unwrap();
+        owned_env
+            .initialize_contract(other_contract_id.clone(), contract_other)
+            .unwrap();
+        owned_env
+            .initialize_contract(self_contract_id.clone(), &contract_self)
+            .unwrap();
 
-    let mut owned_env = OwnedEnvironment::new_max_limit(
-        store.as_clarity_db(&TEST_HEADER_DB, &TEST_BURN_STATE_DB),
-        StacksEpochId::Epoch2_05,
-    );
+        let target_contract = Value::from(PrincipalData::Contract(other_contract_id.clone()));
 
-    owned_env
-        .initialize_contract(trait_contract_id.clone(), contract_trait)
-        .unwrap();
-    owned_env
-        .initialize_contract(other_contract_id.clone(), contract_other)
-        .unwrap();
-    owned_env
-        .initialize_contract(self_contract_id.clone(), &contract_self)
+        eprintln!("{}", &contract_self);
+        execute_transaction(
+            &mut owned_env,
+            p2_principal.clone(),
+            &self_contract_id,
+            "execute",
+            &symbols_from_values(vec![target_contract]),
+        )
         .unwrap();
 
-    let target_contract = Value::from(PrincipalData::Contract(other_contract_id));
+        let (_db, tracker) = owned_env.destruct().unwrap();
+        tracker.get_total()
+    })
+}
 
-    eprintln!("{}", &contract_self);
-    execute_transaction(
-        &mut owned_env,
-        p2_principal,
-        &self_contract_id,
-        "execute",
-        &symbols_from_values(vec![target_contract]),
-    )
-    .unwrap();
+#[test]
+fn test_all() {
+    let baseline = test_tracked_costs("1", StacksEpochId::Epoch20);
 
-    let (_db, tracker) = owned_env.destruct().unwrap();
-    tracker.get_total()
+    for f in NativeFunctions::ALL.iter() {
+        let test = get_simple_test(f);
+        let cost = test_tracked_costs(test, StacksEpochId::Epoch20);
+        assert!(cost.exceeds(&baseline));
+    }
 }
 
 #[test]
-fn test_all() {
-    let baseline = test_tracked_costs("1");
+fn epoch_205_test_all() {
```

Comment:
clarification: What is this testing? What remains to be tested?

Reply:
It tests each individual method's cost function invocation

Reply:
(Added a comment in 2103e9f96)